### DISCUSSION TITLE
Version and Publish with Angular Workspaces

### DISCUSSION THREAD

**Discussion by oliveirarafa** [lerna/lerna#3959] [Q&A]
Guys, I have a monorepo with a stencilJS, react, vue and angular libraries. But following the angular guidelines I had to create an angular workspace, then place my angular library inside the projects folder. 

No problem there, I can build the libraries fine (all depends on the stenciljs lib). But when I need to change the versions and publish, I'm stuck. If I use the `lerna version` and `lerna publish` (both with the correct params) I get the following problems:

- My angular-workspace get the version changed, not my library
- Can't publish since the workspace is private
- My library dependency version (it depends on another from the monorepo) doesn't update
- If I change the workspace to public, I can publish the package, but the name is wrong (it gets the name of the workspace, not the library)

What I might be doing wrong?

**Comment by grantwforsythe** [comment]
Currently facing the same issue following [Stencils official documentation](https://stenciljs.com/docs/angular). The question would be better suited there as it pertains to their documentation but I will add a bit more context with the hopes someone stumbles about this thread with an answer.

The project, managed by Lerna, is structured as below:
```
root/
└── packages
    ├── core/
    └── angular-workspace/
        └── projects/
            └── component-library/
                └── src/
                    ├── lib/
                    └── public-api.ts
```
The core package is public and is a dependency for the component-library defined in the Angular workspace. The workspace is a private package; however, the component-library is public. We would like to version and publish the component-library, not the workspace.

**Reply by oliveirarafa** [reply]
Yeah, I asked there first, but I got an answer to ask here. I'm kind of lost but ended migration to NX (that had the same problems) and created a custom script to publish my library.

**Reply by grantwforsythe** [reply]
I did something similar but I didn't drop Lerna or Nx because I had just refactored my project to use.

I am versioning and publishing the core package with Lerna and I do some gerrymandering to version and publish the Angular library which doesn't feel good to commit as it's beyond messy.

```bash
sed -i "s/\"version\": \"[^\"]*\"/\"version\": \"$(grep -E '"version":' packages/core/package.json | awk -F '"' '{print $4}')\"/g" packages/angular/dist/
component-library/package.json

npm publish packages/angular/dist/component-library
```

Shame on me for following Stencil's official documentation I guess ¯ \ _ ( ツ ) _ / ¯
